.github/workflows/check_interface_compatibility.yml

```diff
@@ -0,0 +1,74 @@
+name: Check interface compatibility
+
+on:
+  pull_request_target:
+
+
+jobs:
+  check-interface-compatibility:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Extract new public interfaces
+        run: |
+          make init
+          bin/public_interface.py extract >> ${{ runner.temp }}/interfaces.new.json
+
+      - name: Backup bin/public_interface.py
+        run: |
+          # Keep a copy of bin/public_interface.py to use on the base branch
+          # So we are using the same bin/public_interface.py to process old/new codebase.
+          cp bin/public_interface.py ${{ runner.temp }}/public_interface.py
+
+      - name: Checkout the base branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
```

Comment:
How about checking out to different directories instead? Currently it's kind of confusing with the `cp`s and such.

Reply:
This approach saves time in `make init`

Reply:
the latest version has enough comment to explain it